scripts/milestone.py

```diff
@@ -0,0 +1,336 @@
+        end_cursor = milestones["pageInfo"]["endCursor"]
+        for edge in milestones["edges"]:
+            if edge["node"]["title"] == title:
+                return edge["node"]["number"]
+        return helper(end_cursor) if end_cursor else None
+
+    return helper()
+
+
+def check_issue(data, problems):
+    # all issues are closed
+    if data["node"]["state"] != "CLOSED":
+        problems.append(f"issue not closed: {description(data)}")
+
+    # all issues have a reso: completed label
+    labels = get_labels(data)
+    if "reso: completed" not in labels:
+        problems.append(f"issue missing resolution: {description(data)}")
+
+    # no issues have more than one reso: label
+    if sum(1 for label in labels if label.startswith("reso:")) > 1:
+        problems.append(f"issue has too many resolutions: {description(data)}")
+
+    # no issues have a status: label
+    if any(label.startswith("status:") for label in labels):
+        problems.append(f"issue has a status: {description(data)}")
+
+    # no issues without a type: label
+    num_types = sum(1 for label in labels if label.startswith("type:"))
+    if num_types == 0:
+        problems.append(f"issue does not have a type: {description(data)}")
+
+    # no issues with multiple type: labels
+    if num_types > 1:
+        problems.append(f"issue has multiple types: {description(data)}")
+
+    # no issues with invalid type: labels
+    if any(label not in VALID_TYPES for label in labels if label.startswith("type:")):
+        problems.append(f"issue has an invalid type: {description(data)}")
+
+    # no issues with TRIAGE label
+    if any(label == "TRIAGE" for label in labels):
+        problems.append(f"issue is in triage: {description(data)}")
+
+
+def check_pr(data, problems):
+    # no unmerged prs
+    if data["node"]["state"] != "MERGED":
+        problems.append(f"PR not merged: {description(data)}")
+
+    # no prs without a status: accepted label
+    labels = get_labels(data)
+    if "status: accepted" not in labels:
+        problems.append(f"PR missing status: {description(data)}")
+
+    # special considerations for "issue prs"
+    if any(label.startswith("type:") for label in labels):
+        # must have a reso: completed label
+        if "reso: completed" not in labels:
+            problems.append(f"issue PR missing resolution: {description(data)}")
+
+        # must not have more than one reso: label
+        if sum(1 for label in labels if label.startswith("reso:")) > 1:
+            problems.append(f"issue PR has too many resolutions: {description(data)}")
+
+    # no prs with more than one status: label
+    if sum(1 for label in labels if label.startswith("status:")) > 1:
+        problems.append(f"PR has too many statuses: {description(data)}")
+
+    # no prs without a type: label
+    num_types = sum(1 for label in labels if label.startswith("type:"))
+    if num_types == 0:
+        problems.append(f"PR does not have a type: {description(data)}")
```

Comment:
I think this block is wrong. PRs *may* have (one) type, those are "issue PRs". But normal PRs linked to a separate issue do not have a type.

Reply:
Fixed! I also had to update the code to account for PRs that don't have a type as you can't sort them with a key based on the value of their type, obviously. I also added a `-a` option to operate on closed milestones.